### DISCUSSION TITLE
Identify player category during race

### DISCUSSION THREAD

**Discussion by mark-inderhees** [braddwalker/ZwiftPacketMonitor#25] [General]
Hi, do you know if player race category (a, b, c, d, e) is advertised via zwift packets or some other mechanism? I've done some digging but come up empty.

**Comment by braddwalker** [comment]
Good question, and unfortunately I'm not entirely sure. There's more payload in the packets than what's currently being deserialized through protobuf, so I'm confident it's in there, but I haven't been able to decode all of it yet.  You can see some of the undefined attributes here. It just takes a bunch of sleuthing to reverse engineer the packets.

https://github.com/braddwalker/ZwiftPacketMonitor/blob/main/src/zwiftMessages.proto

**Reply by mark-inderhees** [reply]
I kind of found it. In player state there is group Id. These are in order of the cats. So if I know my category then I can infer the meaning of group ID.
